loopchain/blockchain/blockchain.py

```diff
@@ -591,8 +591,7 @@ def __add_block(self, block: Block, confirm_info, need_to_write_tx_info=True, ne
             if not need_to_write_tx_info:
                 receipts = None
 
-            if next_prep and self.find_preps_addresses_by_roothash(
-                    Hash32.fromhex(next_prep['rootHash'], ignore_prefix=True)):
+            if next_prep and self.find_preps_addresses_by_roothash(block.header.next_reps_hash):
```

Comment:
For block version 0.3. next_reps_hash is not always filled. See block specifications.  If you want the actual value of next_reps_hash, use reveal_next_reps_hash.